### PR TITLE
[Merged by Bors] - feat(CategoryTheory): the left lifting property is stable under transfinite composition

### PR DESCRIPTION
**Pull request by joelriou**
After #19135 is merged, it will be possible to phrase this in terms of `MorphismProperty.llp` and get better looking statements.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/SmallObject/TransfiniteCompositionLifting.lean [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+variable {F : J ⥤ C} (c : Cocone F) (hc : IsColimit c)
+  {X Y : C} (p : X ⟶ Y)
+  (f : F.obj ⊥ ⟶ X) (g : c.pt ⟶ Y)
+
+/-- Given a colimit cocone `c` for a functor `F : J ⥤ C` from a well-ordered type,
+and maps `p : X ⟶ Y`, `f : F.obj ⊥ ⟶ X`, `g : c.pt ⟶ Y`, this structure
+contains the data of a map `F.obj j ⟶ X` such that `F.map (homOfLE bot_le) ≫ f' = f`
+and `f' ≫ p = c.ι.app j ≫ g`. -/
+@[ext]
+structure SqStruct (j : J) where
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/SmallObject/TransfiniteCompositionLifting.lean:48]: Could you please motivate the name, maybe by drawing an ascii-art diagram in the docstring?

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/CategoryTheory/SmallObject/TransfiniteCompositionLifting.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/SmallObject/TransfiniteCompositionLifting.lean]: Thanks for the suggestion! I have added a comment about the general strategy of the proof and some asciiart.

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**